pkg/reconcilermanager/controllers/reconciler_base.go

```diff
@@ -183,17 +207,22 @@ func (r *reconcilerBase) upsertDeployment(ctx context.Context, reconcilerRef typ
 func (r *reconcilerBase) createOrPatchDeployment(ctx context.Context, declared *appsv1.Deployment) (controllerutil.OperationResult, error) {
 	dRef := client.ObjectKeyFromObject(declared)
 	kind := "Deployment"
-
-	current := &appsv1.Deployment{}
-
-	if err := r.client.Get(ctx, dRef, current); err != nil {
+	forcePatch := true
+	deploymentResourceInterface := r.dynamicClient.Resource(kinds.DeploymentResource()).Namespace(dRef.Namespace)
```

Comment:
`deploymentResourceInterface` -> `deploymentClient` might make this easier to read.

It seems a little weird tho that we have both "deployment" hard-coded here and "deployments" hard-coded in the the kind package. This is what the RESTMapper is for. It maps from kind to resource based on the resources registered in the cluster. We probably shouldn't be hard-coding the resource name.

Reply:
Usually when you inject a Dynamic Client, you also inject a Mapper to use with it.

Reply:
I see, at the beginning I use the RESTMapper and then change it to be like this. I am wondering if I also need to make a fake RESTMapper for our unit test , do you have some suggestions?